L/libjulia/common.jl

```diff
@@ -266,6 +262,24 @@ function build_julia(ARGS, version::VersionNumber; jllversion=version)
         LIBUNWIND:=-losxunwind
         JCPPFLAGS+=-DLIBOSXUNWIND
     EOM
+    else
+        # LLVMLIBUNWIND is currently only defined for USE_SYSTEM_UNWIND=0
+        # but we always need that for apple and julia > 1.6
+        cat << EOM >>Make.user
+        JCPPFLAGS+=-DLLVMLIBUNWIND
+    EOM
```

Comment:
@benlorenz this now contains your PR https://github.com/JuliaLang/julia/pull/55639 but I left the workaround for that issue in as you wrote earlier:
> It might make sense to keep the extra flag here for older julia versions even if the PR is merged

We may wish to at least adjust this comment though, e.g.
```suggestion
    else
        # Ensure LLVMLIBUNWIND is always defined for apple and julia > 1.6
        # to work around issues in old versions of the Julia build system
        # which did not always add this flag when needed
        cat << EOM >>Make.user
        JCPPFLAGS+=-DLLVMLIBUNWIND
    EOM
```

Reply:
I've restricted it to 1.6-1.11 on Apple and tweaked the comment